core/node/da_clients/src/eigenv2m0/client.rs

```diff
@@ -0,0 +1,135 @@
+#[derive(Debug, Clone)]
+pub struct EigenDAClientV2M0 {
+    client: PayloadDisperser,
+}
+
+impl EigenDAClientV2M0 {
+    pub async fn new(config: EigenConfigV2M0, secrets: EigenSecretsV2M0) -> anyhow::Result<Self> {
+        let url = Url::from_str(
+            config
+                .eigenda_eth_rpc
+                .ok_or(anyhow::anyhow!("Eigenda eth rpc url is not set"))?
+                .expose_str(),
+        )
+        .map_err(|_| anyhow::anyhow!("Invalid eth rpc url"))?;
+        let payload_form = match config.polynomial_form {
+            PolynomialForm::Coeff => PayloadForm::Coeff,
+            PolynomialForm::Eval => PayloadForm::Eval,
+        };
+
+        let payload_disperser_config = PayloadDisperserConfig {
+            polynomial_form: payload_form,
+            blob_version: config.blob_version,
+            cert_verifier_address: config.cert_verifier_addr,
+            eth_rpc_url: SecretUrl::new(url),
+            disperser_rpc: config.disperser_rpc,
+            use_secure_grpc_flag: config.authenticated,
+        };
+        let private_key = secrets
+            .private_key
+            .0
+            .expose_secret()
+            .parse()
+            .map_err(|e| anyhow::anyhow!("Failed to parse private key: {}", e))?;
+        let signer = Signer::new(private_key);
+        let client = PayloadDisperser::new(payload_disperser_config, signer)
+            .await
+            .map_err(|e| anyhow::anyhow!("Eigen client Error: {:?}", e))?;
+        Ok(Self { client })
+    }
+}
+
+#[async_trait::async_trait]
+impl DataAvailabilityClient for EigenDAClientV2M0 {
+    async fn dispatch_blob(
+        &self,
+        _: u32, // batch number
+        data: Vec<u8>,
+    ) -> Result<DispatchResponse, DAError> {
+        let payload = Payload::new(data);
+        let blob_key = self
+            .client
+            .send_payload(payload)
+            .await
+            .map_err(to_retriable_da_error)?;
+
+        Ok(DispatchResponse::from(blob_key.to_hex()))
+    }
+
+    async fn ensure_finality(
+        &self,
+        dispatch_request_id: String,
+    ) -> Result<Option<FinalityResponse>, DAError> {
+        // TODO: return a quick confirmation in `dispatch_blob` and await here
+        Ok(Some(FinalityResponse {
+            blob_id: dispatch_request_id,
+        }))
+    }
+
+    async fn get_inclusion_data(&self, blob_id: &str) -> Result<Option<InclusionData>, DAError> {
+        let bytes = hex::decode(blob_id)
+            .map_err(|_| anyhow::anyhow!("Failed to decode blob id: {}", blob_id))
+            .map_err(to_non_retriable_da_error)?;
+        let blob_key = BlobKey::from_bytes(
+            bytes
+                .try_into()
+                .map_err(|_| anyhow::anyhow!("Failed to convert bytes to a 32-byte array"))
+                .map_err(to_non_retriable_da_error)?,
+        );
+        let eigenda_cert = self
+            .client
+            .get_inclusion_data(&blob_key)
+            .await
```

Comment:
I personally would have prefered if the eigenda client remains agnostic to zksync terminology, such that it can be used for other stacks.

Can we rename `client.get_inclusion_data` to `client.get_cert` or something like that? the mapping from cert=inclusion_data should only live in the zksync codebase imo